tar/test/test_option_C_mtree.c

```diff
@@ -33,19 +33,24 @@ DEFINE_TEST(test_option_C_mtree)
 	char *p0;
 	size_t s;
 	int r;
-
 	p0 = NULL;
-
 	char *content = "./foo type=file uname=root gname=root mode=0755\n";
+	char *filename = "output.tar";
 
+	/* an absolute path to mtree file */ 
+	char *mtree_file = "/METALOG.mtree";	
+	char *absolute_path = malloc(strlen(testworkdir) + strlen(mtree_file) + 1);
+	strcpy(absolute_path, testworkdir);
+	strcat(absolute_path, mtree_file );
+	
 	/* Create an archive using an mtree file. */
-	assertMakeFile("/tmp/METALOG.mtree", 0777, content);
+	assertMakeFile(absolute_path, 0777, content);
 	assertMakeDir("bar", 0775);
 	assertMakeFile("bar/foo", 0777, "abc");
 
-	r = systemf("%s -cf output.tar -C bar @/tmp/METALOG.mtree >step1.out 2>step1.err", testprog);
+	r = systemf("%s -cf %s -C bar @%s >step1.out 2>step1.err", testprog, filename, absolute_path);
```

Comment:
Quoting the absolute path argument here will help guard against the possibility that the user's temp directory might have a space in it:
```
systemf("%s -cf %s -C bar \"@%s\" >step1.out 2>step1.err", ...
```

Reply:
Good catch!